### ISSUE TITLE
Formula with table/row references results in #REF when inserting row into a sheet

### ISSUE THREAD

**Issue by keithclarkj** [EPPlusSoftware/EPPlus#1243]
When inserting rows into a sheet that contains a calculated column formula with `[#This Row]` and/or column headers to find data, a #REF is returned replacing `[#This Row]`. 

The formula is ` "=[@value]+[@row]`" in excel.
When referencing the formula, it looks normal:

![image](https://github.com/EPPlusSoftware/EPPlus/assets/77936136/3bc92c6c-ebc5-4abd-b785-4ac16452f388)

But this is the result after inserting rows into the sheet:
![image](https://github.com/EPPlusSoftware/EPPlus/assets/77936136/70823e3f-b238-4f3a-9c9f-bf7ea54921e6)

Calculating the sheet results in a #REF as the result but does not change the formula.

This results in an `InvalidFormulaException` due to the number of closing brackets.

This does not happen when inserting rows via the `ExcelTable` object - only when using `ExcelWorksheet`

The unit test is below (file attached - code will need adjustments if you run of course):

```
[TestMethod]
[DeploymentItem("Resources\\TestTableFormulas.xlsx")]
public void InsertRows_CalculatedColumnFormula()
{
    using (var pck = GetExcelPackage("TestTableFormulas.xlsx"))
    {
        var sheet1 = TryGetWorksheet(pck, "Sheet1");
        var table1 = sheet1.Tables["Table1"];

        var calculatedColumnFormula1 = table1.Columns[3].CalculatedColumnFormula;
        var calculatedColumnFormula2 = table1.Columns[0].CalculatedColumnFormula;

        table1.AddRow(5);
        table1.InsertRow(10, 5);
        sheet1.Calculate();

        sheet1.InsertRow(30, 5);   // throws InvalidFormulaException
    }
}
```

[TestTableFormulas.xlsx](https://github.com/EPPlusSoftware/EPPlus/files/13886514/TestTableFormulas.xlsx)

**Comment by swmal** [comment]
Thanks for reporting, we'll have a look at this.

**Comment by JanKallman** [comment]
Yes, this is a bug. Having a formula with a table address that has a table part and a column reference causes the insert to fail.
I.e.: Table1[[#This Row],[column1]].
We will provide a fix for this.

**Comment by JanKallman** [comment]
Fixed in 7.0.7
